catalog/app/containers/Bucket/requests.js

```diff
@@ -727,20 +717,49 @@ export const listPackages = withErrorHandling(
         today,
         window: analyticsWindow,
       })
-    const packages = await listPackageOwnerPrefixes({ s3, bucket })
-      .then(
-        mapAllP((ownerPrefix) =>
-          listPackagePrefixes({ s3, bucket, ownerPrefix }).then(
-            mapAllP((prefix) =>
-              mergeAllP(
-                fetchPackageLatest({ s3, bucket, prefix }),
-                fetchPackageRevisions({ s3, bucket, prefix }),
-              ),
-            ),
-          ),
-        ),
-      )
-      .then(R.unnest)
+
+    const body = {
+      query: mkFilterQuery(filter),
+      aggs: {
+        packages: {
+          composite: {
+            // the limit is configured in ES cluster settings (search.max_buckets)
+            size: 10000,
```

Comment:
@akarve FYI this is the hard limit on the total number of packages to be displayed in the catalog

Reply:
Do yo mean that you will never issue a request with a size larger than this? Does this affect pagination if the results set is larger than 10K?

Reply:
> Do yo mean that you will never issue a request with a size larger than this?

no, it means ES cant create more than 10k "buckets" (==packages) without adjusting configuration, so we can't paginate further than that

> Does this affect pagination if the results set is larger than 10K?

well, yes and no. the "initial" result set (created by the "composite" aggregation) is sorted by handle A to Z no matter what and truncated to this limit (10k). additional sorting (by timestamp, if enabled) and pagination happens after that.
so, to sum it up, pagination is not affected when using default order (handle A to Z) and may be affected when ordering by timestamp